test/classes/eik-config.test.js

```diff
@@ -291,3 +294,507 @@ test('pathsAndFilesAbsolute - does not throw for directory glob pattern', async
+    const baseDir = join(__dirname, '..', 'fixtures');
+    const config = new EikConfig(
+        {
+            ...validEikConfig,
+            files: {
+                'folder': 'folder/**/*',
+            }
+        },
+        null,
+        baseDir,
+    );
+
+    const mappings = await config.mappings();
+    t.equal(mappings.length, 2);
+    t.equal(mappings[0].source.relative, 'folder/client.js');
+    t.match(mappings[0].source.absolute, 'test/fixtures/folder/client.js');
+    t.equal(mappings[0].destination.packagePathname, '/pkg/pizza/0.0.0');
+    t.equal(mappings[0].destination.filePathname, '/folder/client.js');
+    t.equal(
+        mappings[0].destination.url.href,
+        'http://server/pkg/pizza/0.0.0/folder/client.js',
+    );
+    t.equal(mappings[1].source.relative, 'folder/styles.css');
+    t.match(mappings[1].source.absolute, 'test/fixtures/folder/styles.css');
+    t.equal(mappings[1].destination.packagePathname, '/pkg/pizza/0.0.0');
+    t.equal(mappings[1].destination.filePathname, '/folder/styles.css');
+    t.equal(
+        mappings[1].destination.url.href,
+        'http://server/pkg/pizza/0.0.0/folder/styles.css',
+    );
+
+    t.end();
+});
+
+test('mappings - files is an object - mapped to folder glob - no folder recursion', async (t) => {
+    const baseDir = join(__dirname, '..', 'fixtures');
+    const config = new EikConfig(
+        {
+            ...validEikConfig,
+            files: {
+                'folder': '*',
+            }
+        },
+        null,
+        baseDir,
+    );
+
+    const mappings = await config.mappings();
+    t.equal(mappings.length, 5);
+    t.end();
+});
+
+test('mappings - files is an object - mapped to nested folder', async (t) => {
+    const baseDir = join(__dirname, '..', 'fixtures');
+    const config = new EikConfig(
+        {
+            ...validEikConfig,
+            files: {
+                'path/to/folder': 'folder',
+            }
+        },
+        null,
+        baseDir,
+    );
+
+    const mappings = await config.mappings();
+    t.equal(mappings.length, 2);
+    t.equal(mappings[0].source.relative, 'folder/client.js');
+    t.match(mappings[0].source.absolute, 'test/fixtures/folder/client.js');
+    t.equal(mappings[0].destination.packagePathname, '/pkg/pizza/0.0.0');
+    t.equal(mappings[0].destination.filePathname, '/path/to/folder/client.js');
+    t.equal(
+        mappings[0].destination.url.href,
+        'http://server/pkg/pizza/0.0.0/path/to/folder/client.js',
+    );
+    t.equal(mappings[1].source.relative, 'folder/styles.css');
+    t.match(mappings[1].source.absolute, 'test/fixtures/folder/styles.css');
+    t.equal(mappings[1].destination.packagePathname, '/pkg/pizza/0.0.0');
+    t.equal(mappings[1].destination.filePathname, '/path/to/folder/styles.css');
+    t.equal(
+        mappings[1].destination.url.href,
+        'http://server/pkg/pizza/0.0.0/path/to/folder/styles.css',
+    );
+
+    t.end();
+});
```

Comment:
Some of these tests are perhaps checking the behaviour of `resolvedFiles` method? Perhaps it is worth testing that function directly?

Also I think it would be good to test the error cases for the mapping function as I assume the other methods which have similar logic will be removed along with the tests in the future.